### PR TITLE
Test tableFunctions and tableProcedures in MockConnector

### PR DESCRIPTION
**Pull request by ssheikin**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?
tests

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

> How would you describe this change to a non-technical end user or system administrator?

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by ksobolew — APPROVED** [review]
Nice!

**Comment by ssheikin** [comment]
Something happened with automation here.
Is it common issue? Can we do something about it?

File: testing/trino-tests/src/test/java/io/trino/function/TestingConnectorTableFunction.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import javax.inject.Provider;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.spi.ptf.ReturnTypeSpecification.GenericTable.GENERIC_TABLE;
+import static io.trino.spi.type.BigintType.BIGINT;
+
+public class TestingConnectorTableFunction
```
Comment (kasiafi) [inline testing/trino-tests/src/test/java/io/trino/function/TestingConnectorTableFunction.java:33]: We've got a class `TestingTableFunctions`. Currently it has one TF implementation. You can use te existing implementation or add your implementation there.

**Review by kasiafi — CHANGES_REQUESTED** [review]
This PR is mostly about Table Procedures, as it adds support for testing them with the MockConnector.
Regarding Table Functions, it only tests the MockConnector, not the Table Functions themselves.
Please split the PR into 2 commits accordingly.

**Comment by ssheikin** [comment]
That's because I had to rebase on your latest changes :)

**Review by kokosing — APPROVED** [review]
@kasiafi It looks like comments got addressed. It is all about tests so it is safe to merge. Please let me know if something is missing, @ssheikin will be more than happy to post another PR to address remaining comments. I will merge it to unblock the further work.

**Review by kasiafi — APPROVED** [review]

**Comment by kasiafi** [comment]
CI failure: please update the expected message:
```
[ERROR] io.trino.tests.TestMockConnector.testTableFunction  Time elapsed: 0.052 s  <<< FAILURE!
2022-07-15T18:16:39.7278677Z java.lang.AssertionError: 
2022-07-15T18:16:39.7279090Z 
2022-07-15T18:16:39.7279334Z Expecting message to be:
2022-07-15T18:16:39.7279760Z   <"execution by operator is not yet implemented for table function testing_table_function">
2022-07-15T18:16:39.7280142Z but was:
2022-07-15T18:16:39.7280494Z   <"execution by operator is not yet implemented for table function simple_table_function">
```

**Pull request merged**